### PR TITLE
Move tuple repacking in conversion to DAML-LF into separate function

### PR DESCRIPTION
**Pull request by martin-drhu-da**
I need to use the same logic for implementing the `TextMap.toList` function.

### Pull Request Checklist

- [x] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/master/CONTRIBUTING.md)
- [x] Include appropriate tests
- [x] Set a descriptive title and thorough description
- [ ] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [ ] Add a line to the [release notes](https://github.com/digital-asset/daml/blob/master/docs/source/support/release-notes.rst), if appropriate

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

File: daml-foundations/daml-ghc/src/DA/Daml/GHC/Compiler/Convert.hs [hunk 1]
```diff
@@ -1270,6 +1267,15 @@ mkPure env monad dict t x = do
           `ETyApp` t
           `ETmApp` x
 
+-- | Turn the structural record referenced by `x` into a tuple from `GHC.Tuple`.
```
Comment (neil-da) [inline daml-foundations/daml-ghc/src/DA/Daml/GHC/Compiler/Convert.hs]: Would have been nice to get a little more details about the input structural record invariants in the comment. I assume it's `{_1 : a, _2: b, ...}`?

Comment (martin-drhu-da) [reply daml-foundations/daml-ghc/src/DA/Daml/GHC/Compiler/Convert.hs]: I thought that would be obvious from the `[(FieldName, LF.Type)]` part of the type signature. I'll improve it in a followup PR.

Comment (neil-da) [reply daml-foundations/daml-ghc/src/DA/Daml/GHC/Compiler/Convert.hs]: Maybe it is now I look closer... but I guess i still don't know what the invariants on the field names are.

Comment (martin-drhu-da) [reply daml-foundations/daml-ghc/src/DA/Daml/GHC/Compiler/Convert.hs]: The fields of the input structural record are put into the tuple in the order they appear in the list. The tuple's fields are name `_1`, `_2`, ... as usual. I'll given an example in the code.

**Pull request merged**